src/platform/plugins/shared/discover/public/application/main/state_management/tabs_storage_manager.ts

```diff
@@ -326,23 +375,30 @@ export const createTabsStorageManager = ({
 
     sessionInfo.userId = userId;
     sessionInfo.spaceId = spaceId;
+    sessionInfo.discoverSessionId = persistedDiscoverSession?.id;
 
-    const openTabs = storedTabsState.openTabs.map((tab) => toTabState(tab, defaultTabState));
+    const persistedTabs = persistedDiscoverSession?.tabs.map((tab) =>
+      fromSavedObjectTabToTabState({ tab })
+    );
+    const openTabs =
+      persistedDiscoverSession?.id === storedTabsState.discoverSessionId
+        ? storedTabsState.openTabs.map((tab) => toTabState(tab, defaultTabState))
+        : persistedTabs ?? [];
     const closedTabs = storedTabsState.closedTabs.map((tab) =>
       toRecentlyClosedTabState(tab, defaultTabState)
     );
 
-    if (enabled) {
-      if (selectedTabId) {
-        // restore previously opened tabs
-        if (openTabs.find((tab) => tab.id === selectedTabId)) {
-          return {
-            allTabs: openTabs,
-            selectedTabId,
-            recentlyClosedTabs: closedTabs,
-          };
-        }
+    if (enabled && selectedTabId) {
+      // restore previously opened tabs
+      if (openTabs.find((tab) => tab.id === selectedTabId)) {
+        return {
+          allTabs: openTabs,
+          selectedTabId,
+          recentlyClosedTabs: closedTabs,
+        };
+      }
 
+      if (!persistedDiscoverSession) {
```

Comment:
It seemed complicated to try to handle restoring closed tabs when loading a persisted Discover session and didn't seem necessary, but maybe there's a valid case for it.